tests/src/test/scala/whisk/core/containerpool/test/ContainerPoolTests.scala

```diff
@@ -148,11 +169,43 @@ class ContainerPoolTests extends TestKit(ActorSystem("ContainerPool"))
         pool ! runMessage
         containers(0).expectMsg(runMessage)
         containers(0).send(pool, NeedWork(warmedData()))
+        feed.expectMsg(ContainerReleased)
+        pool ! runMessage2
+        containers(0).expectMsg(Remove)
+        containers(1).expectMsg(runMessage2)
+    }
+
+    it should "remove a container to make space in the pool if it is already full and another action with different invocation namespace arrives" in within(timeout) {
+      val (containers, factory) = testContainers(2)
+      val feed = TestProbe()
+
+      // a pool with only 1 slot
+      val pool = system.actorOf(ContainerPool.props(factory, 1, feed.ref))
+      pool ! runMessage
+      containers(0).expectMsg(runMessage)
+      containers(0).send(pool, NeedWork(warmedData()))
+      feed.expectMsg(ContainerReleased)
+      pool ! runMessage1b
+      containers(0).expectMsg(Remove)
+      containers(1).expectMsg(runMessage1b)
+    }
+
+
+    it should "not remove a container to make space in the pool if it is already full and the same action + same invocation namespace arrives" in within(timeout) {
```

Comment:
below test ends with exectNoMsg() instead of expectMsg() - testing cases where removal result is expected to be None vs Some based on whether this is a different action/invocation namespace